File: LiveTrackingSystem/LiveTracking.Api/Models/OfficeLocation.cs
namespace LiveTracking.Api.Models;

public class OfficeLocation
{
    public int OfficeLocationId { get; set; }
    public string Name { get; set; } = string.Empty;
    public double Latitude { get; set; }
    public double Longitude { get; set; }
    public double RadiusMeters { get; set; } = 100.0;
    public string? Address { get; set; }
    public bool IsActive { get; set; } = true;
    public int? CompanyId { get; set; }
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;

    public Company? Company { get; set; }
    public ICollection<AdminOfficeLocation> AdminOfficeLocations { get; set; } = new List<AdminOfficeLocation>();
}

File: LiveTrackingSystem/MM_Fuel_Apps/app/src/main/java/com/zynexbd/livetracking/models/OfficeLocation.kt
package com.zynexbd.livetracking.models

import com.google.gson.annotations.SerializedName

data class OfficeLocation(
    @SerializedName(value = "id", alternate = ["officeLocationId"]) val id: Int = 0,
    val name: String = "",
    val latitude: Double = 0.0,
    val longitude: Double = 0.0,
    val radiusMeters: Double = 100.0,
    val address: String? = null,
    val isActive: Boolean = true
)

data class CreateOfficeLocationRequest(
    val name: String,
    val latitude: Double,
    val longitude: Double,
    val radiusMeters: Double = 200.0,
    val address: String? = null
)

data class UpdateOfficeLocationRequest(
    val name: String,
    val latitude: Double,
    val longitude: Double,
    val radiusMeters: Double = 200.0,
    val address: String? = null,
    val isActive: Boolean = true
)
